std/datetime.d

```diff
@@ -29161,6 +29191,74 @@ private:
         _hasDST = hasDST;
     }
 
+    // Android concatenates the usual timezone directories into a single file,
+    // tzdata, along with an index to jump to each timezone's offset.  In older
+    // versions of Android, the index was stored in a separate file, zoneinfo.idx,
+    // whereas now it's stored at the beginning of tzdata.
+    version(Android)
+    {
+        // Keep track of whether there's a separate index, zoneinfo.idx.  Only
+        // check this after calling tzdataIndex, as it's initialized there.
+        static shared bool separate_index;
+
+        // Extracts the name of each time zone and the offset where its data is
+        // located in the tzdata file from the index and caches it for later.
+        static uint[string] tzdataIndex(string tzDir)
+        {
+            import std.concurrency : initOnce;
+            import std.conv : to;
+            import std.format : format;
+            import std.path : asNormalizedPath, chainPath;
+
+            static __gshared uint[string] _tzIndex;
```

Comment:
I have no idea if AAs are thread safe. If it turns they are thread safe on lookup only, then maybe `tzdataIndex` can enforce this by returning `const(uint[string])` so all the "dangerous" stuff is contained.